### ISSUE TITLE
SSH error: "The output was:\n"

### ISSUE THREAD

**Issue by Jeffdude** [ansible/ansible#11150]
I am getting a very strange error when running my ansible playbook on some newly created openstack vms. It seems that ansible's ssh is throwing an error, while I can still ssh using my private key and also by using the `nova ssh` command. The error is very short and non descriptive, so please let me know if you need to see some other files from me, since I am a little new and not quite sure how to make this error more verbose. Here is the error:

```
TASK [os_provision : add facts from openstack] **********************************
ok: [mesos-slave-2] => {"changed": false, "ansible_facts": {"ansible_ssh_host": "<redacted>", "ansible_ssh_private_key_file": "keys/ccs-os.pem"}}
ok: [mesos-slave-3] => {"changed": false, "ansible_facts": {"ansible_ssh_host": "<redacted>", "ansible_ssh_private_key_file": "keys/ccs-os.pem"}}
ok: [master-1] => {"changed": false, "ansible_facts": {"ansible_ssh_host": "<redacted>", "ansible_ssh_private_key_file": "keys/ccs-os.pem"}}
ok: [mesos-slave-1] => {"changed": false, "ansible_facts": {"ansible_ssh_host": "<redacted>", "ansible_ssh_private_key_file": "keys/ccs-os.pem"}}

PLAY: ***************************************************************************

TASK [setup] ********************************************************************
fatal: [master-1]: FAILED! => {"msg": "ERROR! SSH encountered an unknown error. The output was:\n", "failed": true}
fatal: [mesos-slave-1]: FAILED! => {"msg": "ERROR! SSH encountered an unknown error. The output was:\n", "failed": true}
fatal: [mesos-slave-2]: FAILED! => {"msg": "ERROR! SSH encountered an unknown error. The output was:\n", "failed": true}
fatal: [mesos-slave-3]: FAILED! => {"msg": "ERROR! SSH encountered an unknown error. The output was:\n", "failed": true}

NO MORE HOSTS LEFT **************************************************************
PLAY RECAP **********************************************************************
master-1                   : ok=3    changed=1    unreachable=0    failed=1
mesos-slave-1              : ok=3    changed=1    unreachable=0    failed=1
mesos-slave-2              : ok=3    changed=1    unreachable=0    failed=1
mesos-slave-3              : ok=3    changed=1    unreachable=0    failed=1
```

Here is some system info. This is very fresh CentOS 7 vm.

```
[jeff@astraea ansible]$ uname -a
Linux astraea 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
[jeff@astraea ansible]$ ansible --version
ansible 2.0.0
  configured module search path = /usr/share/ansible:./modules
[jeff@astraea ansible]$ pip show shade

---
Metadata-Version: 2.0
Name: shade
Version: 0.6.4
Summary: Client library for operating OpenStack clouds
Home-page: http://docs.openstack.org/infra/shade/
Author: OpenStack Infrastructure Team
Author-email: openstack-infra@lists.openstack.org
License: UNKNOWN
Location: /usr/lib/python2.7/site-packages
Requires: python-glanceclient, python-ironicclient, os-client-config, six, python-swiftclient, bunch, python-troveclient, pbr, dogpile.cache, python-cinderclient, jsonpatch, python-novaclient, python-keystoneclient, python-neutronclient
[jeff@astraea ansible]$ pip show python-novaclient

---
Metadata-Version: 2.0
Name: python-novaclient
Version: 2.25.0
Summary: Client library for OpenStack Compute API
Home-page: https://git.openstack.org/cgit/openstack/python-novaclient
Author: OpenStack
Author-email: openstack-dev@lists.openstack.org
License: Apache License, Version 2.0
Location: /usr/lib/python2.7/site-packages
Requires: oslo.serialization, oslo.i18n, pbr, argparse, requests, PrettyTable, iso8601, Babel, six, oslo.utils, simplejson, python-keystoneclient
```

**Comment by bcoca** [comment]
can you show the failing task definitions?

**Comment by Jeffdude** [comment]
Sure, though I don't think it is failing because of the specific task definitions. I think it is because it is going from local execution to remote execution, and without being able to ssh, ansible fails. Here is the task it fails on:

```
- hosts: all
  sudo: yes
  user: centos


  pre_tasks:
    - name: remove extraneous hosts entries
      copy:
        dest: /etc/hosts
        content: |
          127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
          ::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
   - name: turn selinux off
      selinux: policy=targeted state=permissive

    - name: "Build hosts file"
      lineinfile:
        dest: /etc/hosts
        regexp: '.*{{ item }}$'
        line: "{{ openstack.addresses['2015IPCM'][addr] }} {{'item'}}"
        state: present
      with_items: groups['all']
    - debug: var=openstack.addresses['2015IPCM'][addr]
    - pause: prompt="see address"
```

And the error is thrown before The debug var is printed or the pause task is executed. Thanks for your help.

**Comment by bcoca** [comment]
this might be related to #11132

**Comment by jtktam** [comment]
it is the same behavior as #11132

**Comment by Jeffdude** [comment]
Okay, I managed to fix this issue by downgrading to Ansible 1.9.1. The switch required going back from the os_server to nova_compute module. After that, my definition of `ansible_ssh_user: centos` works and my playbook can finally run. So I believe this is something that was introduced in the 2.0 release.

**Comment by bcoca** [comment]
confirmed then, this is dupe of  #11132
